source/index.js

```diff
@@ -54,9 +54,7 @@
 			readdir: function(path, callback) {
 				client.getDir(endpoint, path)
 					.then(function(dirData) {
-						(callback)(null, dirData.map(function(dirEntry) {
-							return dirEntry.basename;
-						}));
+						(callback)(null, dirData);
```

Comment:
I do agree, for performance reasons, that it's a problem. It needs to be (in my opinion) compatible with output from fs.readdir as well as negating the need to stat each file contained therein.

I'm open to suggestions - I'll do some thinking.